examples/cms-sanity/pages/api/preview.js

```diff
@@ -23,6 +26,6 @@ export default async function preview(req, res) {
 
   // Redirect to the path from the fetched post
   // We don't redirect to req.query.slug as that might lead to open redirect vulnerabilities
-  res.writeHead(307, { Location: `/posts/${post.slug}` })
+  res.writeHead(307, { Location: `/posts/${req.query.slug}` })
```

Comment:
Note the comment above.
```suggestion
  res.writeHead(307, { Location: `/posts/${post.slug}` })
```

Reply:
Ooops 😓 Will change this back.
How can this be exploited, though? We're fetching a post by the slug passed in the query and return early if a post with the given slug does not exist. Unless I'm mistaken, `post.slug` will always be equal to `req.query.slug` here?

Reply:
Fixed - but would still like to understand how the open redirect vulnerability could happen here